### PR TITLE
First http request to Cromwell Server

### PR DESCRIPTION
**Pull request by lmtani**
## Description

This PR adds query feature (`/api/workflows/v1/query`) to our CLI. The main objective here is to create a pattern to organize request/response struct and the functions we'll use to interact with the server.

## Details

- A type (`Client`) was created to store functions related to HTTP requests
- The template of several Client functions was added.
- The first steps for 'QueryWorkflow' (command "query") are done.
- .gitignore changed from Python to Go

## Test

Follow the README.md **Development environment** instructions.

### CODE REVIEW

**Review by marcosinger — COMMENTED** [review]
I did some suggestions to make the code more idiomatic and easier to follow. Let me know if something is not that clear.

I notice that submitting a job as the `README` says, generate an error on my side:

### Job
```
+--------------------------------------+------+-------------------+-------------------+-----------+
|              OPERATION               | NAME |       START       |        END        |  STATUS   |
+--------------------------------------+------+-------------------+-------------------+-----------+
| 8d98881b-b588-4f3f-ae88-c404e6b9490a |      | 0001-01-01 00h00m | 0001-01-01 00h00m | Submitted |
+--------------------------------------+------+-------------------+-------------------+-----------+
```

### Query

```
~ ./cromwell q
+--------------------------------------+------------+-------------------+-------------------+---------+
|              OPERATION               |    NAME    |       START       |        END        | STATUS  |
+--------------------------------------+------------+-------------------+-------------------+---------+
| 8d98881b-b588-4f3f-ae88-c404e6b9490a | HelloWorld | 2020-09-16 18h08m | 0001-01-01 00h00m | Running |
+--------------------------------------+------------+-------------------+-------------------+---------+
```

```
~ ./cromwell q
+--------------------------------------+------------+-------------------+-------------------+--------+
|              OPERATION               |    NAME    |       START       |        END        | STATUS |
+--------------------------------------+------------+-------------------+-------------------+--------+
| 8d98881b-b588-4f3f-ae88-c404e6b9490a | HelloWorld | 2020-09-16 18h08m | 2020-09-16 18h08m | Failed |
+--------------------------------------+------------+-------------------+-------------------+--------+
```

File: commands/server_client.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+func (c *Client) Kill(o string) (SubmitResponse, error) {
+	route := fmt.Sprintf("/api/workflows/v1/%s/abort", o)
+	r, err := c.post(route, map[string]string{})
+	if err != nil {
+		return SubmitResponse{}, err
+	}
+	defer r.Body.Close()
+	body, _ := ioutil.ReadAll(r.Body)
```
[the remark this answers is not in the document]
Comment (lmtani) [reply commands/server_client.go:79]: I think this way is more cleaner. I'm going to use it.

File: commands/server_client.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+func (c *Client) post(u string, files map[string]string) (*http.Response, error) {
+	uri := fmt.Sprintf("%s%s", c.host, u)
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+	for k, v := range files {
+		file, _ := os.Open(v)
+		content, _ := ioutil.ReadAll(file)
+		fi, _ := file.Stat()
```
[the remark this answers is not in the document]
Comment (lmtani) [reply commands/server_client.go:55]: Errors? where? :see_no_evil:

**Comment by lmtani** [comment]
@marcosinger  Thank you for the review! 

The failure in your workflow execution may be related to the way Cromwell uses docker. I'm not sure about how it launches containers if your user is not in the docker group.

I'll remove the docker from the WDL runtime section, so it should work without docker for macOS and Linux users. Probably it is more common people not having Docker than running this example on Windows.

File: commands/get.go [hunk 3]
```diff
@@ -2,23 +2,45 @@ package commands
...
+	timePattern := "2006-01-02 15h04m"
+	for _, elem := range workflows.Results {
+		rows = append(rows, []string{elem.ID, elem.Name, elem.Start.Format(timePattern), elem.End.Format(timePattern), elem.Status})
 	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
+	return header, rows
+}
+
+func QueryWorkflow(c *cli.Context) error {
```
[the remark this answers is not in the document]
Comment (lmtani) [reply commands/get.go:36]: I think I liked it, but I'll keep this suggestion here so I cant return to this later. I'm in module two of [this coursera](https://www.coursera.org/specializations/google-golang) course about golang and soon interfaces will be presented.

**Pull request merged**